ReactCommon/cxxreact/JSBigString.cpp

```diff
@@ -14,6 +14,59 @@
 namespace facebook {
 namespace react {
 
+JSBigFileString::JSBigFileString(int fd, size_t size, off_t offset /*= 0*/)
+  : m_fd { -1 }
+  , m_data { nullptr } {
+  folly::checkUnixError(m_fd = dup(fd),
+    "Could not duplicate file descriptor");
+
+  // Offsets given to mmap must be page aligend. We abstract away that
+  // restriction by sending a page aligned offset to mmap, and keeping track
+  // of the offset within the page that we must alter the mmap pointer by to
+  // get the final desired offset.
+  if (offset != 0) {
+    const static auto ps = getpagesize();
+    auto d = lldiv(offset, ps);
+
+    m_mapOff = d.quot;
+    m_pageOff = d.rem;
+    m_size = size + m_pageOff;
+  } else {
+    m_mapOff = 0;
+    m_pageOff = 0;
+    m_size = size;
+  }
+}
+
+JSBigFileString::~JSBigFileString() {
+  if (m_data) {
+    munmap((void *)m_data, m_size);
+  }
+  close(m_fd);
+}
+
+
+const char *JSBigFileString::c_str() const {
+  if (!m_data) {
+    m_data =
+      (const char *) mmap(0, m_size, PROT_READ, MAP_PRIVATE, m_fd, m_mapOff);
```

Comment:
`#include <folly/portability/SysMMan.h` and change `MAP_PRIVATE` to `MAP_SHARED` (this is a read-only `mmap`, so copy-on-write is irrelevant) and this should work just fine on Windows :)